tests/steamship_tests/app/integration/test_request_id.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+from steamship_tests import PACKAGES_PATH, PLUGINS_PATH
+from steamship_tests.utils.deployables import deploy_package, deploy_plugin
+
+from steamship import Steamship
+
+
+@pytest.mark.usefixtures("client")
+def test_get_param_decoding(client: Steamship):
+    request_id_package_path = PACKAGES_PATH / "request_id_plumbing_test_package.py"
+    request_id_plugin_path = PLUGINS_PATH / "generators" / "request_id_generator.py"
+
+    with deploy_package(client, request_id_package_path) as (package, version, instance):
+        with deploy_plugin(client, request_id_plugin_path, plugin_type="generator") as (
+            plugin,
+            _,
+            _,
+        ):
+            result = instance.invoke("requestids", plugin_handle=plugin.handle)
+            package_request_id = result.get("packageRequestId")
```

Comment:
Not necessary in this PR, but an awesome addition would be to have the Proxy stamp a header on all requests that contained the requestId.

Then in addition to `assert package_requestId is not None` we could check against the Proxy's opinion of events.

I guess that would require us to use a raw HTTP post instead of `instance.invoke` to test though

Reply:
This is a good idea; should be very easy to add to the engine PR